api/nodes/serializers.py

```diff
@@ -124,6 +127,123 @@ def update(self, instance, validated_data):
         return instance
 
 
+class NodeContributorsSerializer(JSONAPISerializer):
+    """ Separate from UserSerializer due to necessity to override almost every field as read only
+    """
+    filterable_fields = frozenset([
+        'fullname',
+        'given_name',
+        'middle_name',
+        'family_name',
+        'id',
+        'bibliographic',
+        'permissions'
+    ])
+    id = ser.CharField(source='_id')
+    fullname = ser.CharField(read_only=True, help_text='Display name used in the general user interface')
+    given_name = ser.CharField(read_only=True, help_text='For bibliographic citations')
+    middle_name = ser.CharField(read_only=True, source='middle_names', help_text='For bibliographic citations')
+    family_name = ser.CharField(read_only=True, help_text='For bibliographic citations')
+    suffix = ser.CharField(read_only=True, help_text='For bibliographic citations')
+    date_registered = ser.DateTimeField(read_only=True)
+    gravatar_url = ser.CharField(read_only=True, help_text='URL for the icon used to identify the user. '
+                                                           'Relies on http://gravatar.com ')
+    social_accounts = ser.DictField(read_only=True, source='social',
```

Comment:
Is there any way to do bulk queries on the user endpoint, for cases when the query wants full user data on every  contributor? (perhaps filter params) 

It seems that the central issue is that user data is duplicated across contribs and users; if contribs was just a list of userid, permission, and bibliographic status then the duplicate code would go away. Haven't read requirements in depth though.